src/goto-instrument/contracts/contracts.h

```diff
@@ -210,6 +209,23 @@ class code_contractst
     codet &expression,
     source_locationt location,
     const irep_idt &mode);
+
+  /// \brief Finds symbols declared with a static lifetime in the given
+  /// `root_function` or one of the functions it calls,
+  /// and adds them to the given `assigns`.
+  ///
+  /// @param functions all functions of the goto_model
+  /// @param root_function the root function under which to search statics
+  /// @param assigns assigns clause where search results are added
+  ///
+  /// A symbol is considered a static local symbol iff:
+  /// - it has a static lifetime annotation
+  /// - its source location has a non-empty function attribute
+  /// - this function attribute is found in the call graph of the root_function
+  void find_static_locals(
```

Comment:
[nit] this function isn't just "finding" static locals, it's also adding them to `assigns`. It's worth thinking about:
1. should we make this function "pure": only find static locals and return a `set` of those?
2. if not, would it make sense to move this function into the `assigns_clauset` class as a member function?
3. if not, should we rename this to `add_static_locals_to_assigns_set`?

Reply:
I moved the method to the `assigns_clauset` class.